### ISSUE TITLE
spark_apply  returns Error in file(con, "r").... Permission denied

### ISSUE THREAD

**Issue by JannicCutura** [rstudio/sparklyr#2034]
# Reporting an Issue with sparklyr

For general programming questions with `sparklyr`, please ask on
[Stack Overflow](http://stackoverflow.com) instead.

Please briefly describe your problem and, when relevant, the output you expect.
Please also provide the output of `utils::sessionInfo()` or
`devtools::session_info()` at the end of your post.

If at all possible, please include a [minimal, reproducible
example](https://stackoverflow.com/questions/5963269/how-to-make-a-great-r-reproducible-example).
The `sparklyr` team will be much more likely to resolve your issue if they are
able to reproduce it themselves locally. If possible, try generating a reproducible
example using the [reprex](https://github.com/tidyverse/reprex) package.

Please delete this preamble after you have read it.

---

I wanted to add leading zeros in sparklyr using spark_apply. However I run into a strange error, when executing the last line of the code below:

`Error in file(con, "r") : cannot open the connection
In addition: Warning message:
In file(con, "r") :
  cannot open file 'C:\Users\janni\AppData\Local\Temp\Rtmp4KoLZV\file710677e2eef_spark.log': Permission denied`

```r
spark_version <- "2.1.0"
sc <- spark_connect(master = "local", version = spark_version)
df <- data.frame(id = c(1, 1, 2, 2), county_code = c(1, 20, 321, 2))
sprintf("%03d",as.numeric(df$county_code))

df_tbl = copy_to(sc,df, "df_tbl", overwrite = TRUE)

df_tbl %>% spark_apply(function(e) data.frame(sprintf("%03d",as.numeric(e$county_code), e),
                                              names = c('county_code_fips', colnames(e))))

```

**Comment by nik-maheshwari** [comment]
I have ran into this error before. From what I can recall, I restarted my R session or started RStudio in Admin mode. Can you try these and see if it helps?

**Comment by gwd666** [comment]
No, not even running the github sparklyr 1.1.0.9001 version in RStudio as Admin, which imho is not an acceptable solution for this problem anyway, solves this issue, which also shows up if you execute the code examples from https://github.com/sparklyr/sparklyr (section "Distributed R") 
This works: 
`spark_apply(iris_tbl, function(data) {
  data[1:4] + rgamma(1,2)
})`      
This throws the above error: 
`spark_apply(
  iris_tbl,
  function(e) broom::tidy(lm(Petal_Width ~ Petal_Length, e)),
  columns = c("term", "estimate", "std.error", "statistic", "p.value"),
  group_by = "Species"
)`

**Comment by gwd666** [comment]
One more important note - the above sample code (Distributed R: spark_apply)   
```
spark_apply(     
  iris_tbl,     
  function(e) broom::tidy(lm(Petal_Width ~ Petal_Length, e)),     
  columns = c("term", "estimate", "std.error", "statistic", "p.value"),     
  group_by = "Species"     
)
```
runs without any errors on Ubuntu 18.04 (WSL2 on WINDOWS 10).
